### ISSUE TITLE
td-has-headers takes 30+ seconds

### ISSUE THREAD

**Issue by patrickhulce** [dequelabs/axe-core#908]
1. Open https://www.guidetoonlineschools.com/online-schools
2. Add axe 3.0.2 to the page (https://cdnjs.cloudflare.com/ajax/libs/axe-core/3.0.2/axe.js)
3. Run td-has-headers rule in console
```js
await window.axe.run(document, {
  elementRef: true,
  runOnly: {
    type: 'rule',
    values: ['td-has-header'],
  },
  resultTypes: ['violations', 'inapplicable'],
})
```
4. Wait...

![image](https://user-images.githubusercontent.com/2301202/40384097-807f597a-5db7-11e8-86fb-4c23ab12f541.png)

Seems like you might want to memoize toGrid or something, O(w^2*h^2) is rough :)
https://github.com/dequelabs/axe-core/blob/cd1e5eee6f7d9c354986799ea2093f9649b830f7/lib/commons/table/get-headers.js#L17

Originally reported in https://github.com/GoogleChrome/lighthouse/issues/5249, Lighthouse axe run never completes within 60s

<pre><code>
axe-core version: 3.0.2
chrome version: 66.0.3359.181

</code></pre>

**Comment by paulirish** [comment]
http://giveawaylisting.com/ takes over 3 minutes due to this issue.

**Comment by ggennrich** [comment]
Any update on this issue? We have a single page on our website that breaks due to this issue. Could someone explain in detail what exactly is breaking in lighthouse so that I have a better idea of what's going on? The fact this breaks on just one of our pages leads me to believe there is something unusual on the page. I'm worried this may be impacting our rankings as we noticed some drops in the last couple months.

**Comment by WilcoFiers** [comment]
I'm not aware of anyone working on this issue. If you've got some time, we'd very much welcome some suggestions on how to improve the performance. A pull request would also be fantastic. Marking this as help wanted, as I don't see us prioritising this issue any time soon.

**Comment by padmavemulapati** [comment]
Not taking more than 2seconds for me, verified for axe-core:3.0.3 and chrome 78 version
